File: test90_out/local.output/SendOsc/SendOsc.c

/*
 * ======== string literals ========
 */

static const char local_output_SendOsc__s__fileName[] = "SendOsc.p";
static const char local_output_SendOsc__s__1[] = "SendOsc";
static const char local_output_SendOsc__s__0[] = "oscsend 127.0.0.1 7000 /led/1 iii ";

/*
 * ======== code injections ========
 */

 #include <stdlib.h> 

/*
 * ======== function definitions ========
 */

void local_output_SendOsc_sendRGB__E( uint8 r, uint8 g, uint8 b ) {
    byte tmp[5];
    string t;
    uint8 len;
    
    t = (tmp);
    len = 0;
    local_output_SendOsc_sprint__I((local_output_SendOsc_header__V), (local_output_SendOsc_data__A), 0);
    pollen_text_Text_itoa__E((r), (tmp), 10);
    local_output_SendOsc_sprint__I((t), (local_output_SendOsc_data__A), pollen_text_Text_strlen__E((local_output_SendOsc_header__V)));
    (len) = pollen_text_Text_strlen__E((local_output_SendOsc_data__A));
    (local_output_SendOsc_data__A[(len)]) = ' ';
    (local_output_SendOsc_data__A[(len) + 1]) = '\0';
    pollen_text_Text_itoa__E((g), (tmp), 10);
    local_output_SendOsc_sprint__I((t), (local_output_SendOsc_data__A), pollen_text_Text_strlen__E((local_output_SendOsc_data__A)));
    (len) = pollen_text_Text_strlen__E((local_output_SendOsc_data__A));
    (local_output_SendOsc_data__A[(len)]) = ' ';
    (local_output_SendOsc_data__A[(len) + 1]) = '\0';
    pollen_text_Text_itoa__E((b), (tmp), 10);
    local_output_SendOsc_sprint__I((t), (local_output_SendOsc_data__A), pollen_text_Text_strlen__E((local_output_SendOsc_data__A)));
     system(local_output_SendOsc_data__A) ;

}

void local_output_SendOsc_sprint__I( string s, byte  data[], uint16 start ) {
    uint16 i;
    byte * b = (char*) (s);
    i = 0;
    while ((b[(i)]) != '\0') {
        (data[(start) + (i)]) = (b[(i)]);
        (i)++;
    }
    (data[(start) + (i)]) = '\0';
}

void local_output_SendOsc_targetInit__I() {
}

File: test90_out/local.output/SendOsc/SendOsc.js

/*
 * ======== MODULE local.output.SendOsc ========
 */

var $$c = function() {
    const SendOsc = {};
    SendOsc.SendOsc = SendOsc
    SendOsc.$name = 'local.output.SendOsc';
    SendOsc.pollen$used = 0;
    SendOsc.Text = $units['pollen.text.Text'];
    SendOsc.sendRGB = new $$Ref('local_output_SendOsc_sendRGB__E');
    SendOsc.sprint = new $$Ref('local_output_SendOsc_sprint__I');
    SendOsc.targetInit = new $$Ref('local_output_SendOsc_targetInit__I');
    SendOsc.$$hostInit = function() {
        var $$text = '';
        SendOsc.data = new $$Array(50, function($$cn,$$idx){ return 0;}, 'local_output_SendOsc_data__A', false); SendOsc.data.$$assign((new $$Array(50, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ]));
        return $$text;
    }
    SendOsc.pollen__uses__ = function() {
        $$bind($units['pollen.text.Text'], 'pollen$used', true);
    }
    SendOsc.$$privateInit = function() {
        SendOsc.header = "oscsend 127.0.0.1 7000 /led/1 iii ";
        SendOsc.data = new $$Array(50, function($$cn,$$idx){ return 0;}, 'local_output_SendOsc_data__A', false); SendOsc.data.$$assign((new $$Array(50, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ]));
    }
    return SendOsc;
}
var $$u = $$c();
$units['local.output.SendOsc'] = $$u;
$units.push($$u);

